### PR TITLE
kernel: bump 6.1 to 6.1.40

### PR DESCRIPTION
**Pull request by graysky2**
kernel: bump 6.1 to 6.1.40

Removed upstreamed:
	generic/backport-6.1/701-v6.5-net-bgmac-postpone-turning-IRQs-off-to-avoid-SoC-han.patch[1]
	generic/pending-6.1/160-workqueue-fix-enum-type-for-gcc-13.patch[2]
	qualcommax/patches-6.1/0022-v6.5-soc-qcom-mdt_loader-Fix-unconditional-call-to-scm_pa.patch[3]

Manually rebased:
	bcm27xx/patches-6.1/950-0359-xhci-quirks-add-link-TRB-quirk-for-VL805.patch
	bcm27xx/patches-6.1/950-0362-usb-xhci-add-VLI_TRB_CACHE_BUG-quirk.patch
	bcm27xx/patches-6.1/950-0390-usb-xhci-add-a-quirk-for-Superspeed-bulk-OUT-transfe.patch
	bcm27xx/patches-6.1/950-0469-usb-xhci-add-XHCI_VLI_HUB_TT_QUIRK.patch
	bcm53xx/patches-6.1/180-usb-xhci-add-support-for-performing-fake-doorbell.patch

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.40&id=685b57a1221c38ec8b456f968264d2496715820c
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.40&id=2d57a1590f4d8c516f5aaf8fd5bb4f52d67275d8
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.40&id=33f8dff6e1cbba5c2ec85fa5649c0a759a7e685c

Build system: x86/64
Build-tested: x86/64, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: x86/64, filogic/xiaomi_redmi-router-ax6000-ubootmod

### CODE REVIEW

**Comment by graysky2** [comment]
@hauke @PolynomialDivision Please feel free to use this as a start, but this one is over my head (bcm27xx and mediatek).

**Comment by PolynomialDivision** [comment]
@graysky2 You write in your commit message, that you removed `950-0034-drm-bridge-Introduce-pre_enable_upstream_first-to-al.patch` (that's correct it got upstreamed) but did you forget to commit that removal?

**Comment by graysky2** [comment]
Sorry about that, yes.  Fixed.

**Comment by hauke** [comment]
I pushed a commit which refreshes the mediatek and bcm27xx target. Feel free to squash it into the original commit.

**Comment by graysky2** [comment]
@hauke 
> I pushed a commit which refreshes the mediatek and bcm27xx target. Feel free to squash it into the original commit.

I looked at your fork but did not see a new commit.  Can you advise?

**Comment by graysky2** [comment]
ping @hauke - can you share the location of the commit you pushed?

**Comment by DragonBluep** [comment]
@graysky2 hauke pushed the patch to your graysky2:6.1 branch in this PR, but you accidently override it by force-push. You can download it from here.
https://github.com/openwrt/openwrt/commit/c3a4498b4b5da7c77c707037b51658bff3113603.patch

**Comment by graysky2** [comment]
Thanks @DragonBluep - I did not see his message until after I did the force push.  I need to get into the habit of something like:
```
git pull --rebase && git push origin +$(git rev-parse --abbrev-ref HEAD)
```

**Comment by graysky2** [comment]
@DragonBluep - how did you manage to locate the commit hash of his push after I did the force push?

**Comment by DragonBluep** [comment]
You can click here to enter into the commit page and add `.patch` suffix to the URL to download the patch.
image

**Comment by graysky2** [comment]
@hauke - we're getting a build fail on bcm27xx.  I have the 6.1.40 push ready but am thinking we want 6.1.39 to be ironed out first.

**Comment by Ansuel** [comment]
@graysky2 took care of it just now ;)

**Comment by graysky2** [comment]
> @graysky2 took care of it just now ;)

Thanks, Christian.  Did you commit only change a description?  I already processed 6.1.40 based on this from a few hours ago.

**Comment by Ansuel** [comment]
> > @graysky2 took care of it just now ;)
> 
> Thanks, Christian. Did you commit only change a description? I already processed 6.1.40 based on this from a few hours ago.

i refactored the description a bit and had to fix 2 patch for bcm27xx but in theory you should be able to rebase 6.1.40 on top of this with now problem. Also I fixed the author that was wrong... (no idea what happen there)

**Comment by Ansuel** [comment]
> bcm53xx/patches-6.1/180-usb-xhci-add-support-for-performing-fake-doorbell.patch
>         bcm27xx/patches-6.1/950-0359-xhci-quirks-add-link-TRB-quirk-for-VL805.patch
>         bcm27xx/patches-6.1/950-0362-usb-xhci-add-VLI_TRB_CACHE_BUG-quirk.patch
>         bcm27xx/patches-6.1/950-0390-usb-xhci-add-a-quirk-for-Superspeed-bulk-OUT-transfe.patch
>         bcm27xx/patches-6.1/950-0469-usb-xhci-add-XHCI_VLI_HUB_TT_QUIRK.patch

the changed patch doesn't affect usb so it should not be needed

**Comment by graysky2** [comment]
OK it probably makes sense for me to just repeat the run and see if any of the 5 patches I called out above in my edit are different as a function of your work.

**Comment by graysky2** [comment]
> the changed patch doesn't affect usb so it should not be needed

OK :+1:

**Comment by Ansuel** [comment]
@graysky2 (just in case you have pending change to push and you have problem with refreshing/rebase, you can create a new branch with your local changes committed. (git checkout -b temp) Then fetch origin (git fetch origin) and reset the 6.1 branch to this new version (git reset --hard origin/6.1) and then cherry-pick or merge the temp branch on top of this (git merge temp))

File: target/linux/bcm27xx/patches-6.1/950-0390-usb-xhci-add-a-quirk-for-Superspeed-bulk-OUT-transfe.patch [hunk 1]
```diff
@@ -86,11 +86,15 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
...
- #define XHCI_AVOID_DQ_ON_LINK	BIT_ULL(45)
- #define XHCI_VLI_TRB_CACHE_BUG	BIT_ULL(46)
-+#define XHCI_VLI_SS_BULK_OUT_BUG	BIT_ULL(47)
+ #define XHCI_AVOID_DQ_ON_LINK  BIT_ULL(45)
+ #define XHCI_VLI_TRB_CACHE_BUG BIT_ULL(46)
+-#define XHCI_ZHAOXIN_TRB_FETCH	BIT_ULL(47)
+-#define XHCI_ZHAOXIN_HOST	BIT_ULL(48)
++#define XHCI_VLI_SS_BULK_OUT_BUG       BIT_ULL(47)
++#define XHCI_ZHAOXIN_TRB_FETCH	BIT_ULL(48)
++#define XHCI_ZHAOXIN_HOST	BIT_ULL(49)
```
Comment (hauke) [inline target/linux/bcm27xx/patches-6.1/950-0390-usb-xhci-add-a-quirk-for-Superspeed-bulk-OUT-transfe.patch:97]: Please move `XHCI_VLI_SS_BULK_OUT_BUG` again to the end

**Comment by Ansuel** [comment]
@graysky2 yep much better less patch delta for me you can squash to the .40 commit

File: target/linux/bcm53xx/patches-6.1/180-usb-xhci-add-support-for-performing-fake-doorbell.patch [hunk 2]
```diff
@@ -107,11 +107,15 @@ it on BCM4708 family.
...
+@@ -1899,8 +1899,9 @@ struct xhci_hcd {
  #define XHCI_EP_CTX_BROKEN_DCS	BIT_ULL(42)
  #define XHCI_SUSPEND_RESUME_CLKS	BIT_ULL(43)
  #define XHCI_RESET_TO_DEFAULT	BIT_ULL(44)
-+#define XHCI_FAKE_DOORBELL	BIT_ULL(45)
+-#define XHCI_ZHAOXIN_TRB_FETCH	BIT_ULL(45)
+-#define XHCI_ZHAOXIN_HOST	BIT_ULL(46)
++#define XHCI_FAKE_DOORBELL     BIT_ULL(45)
++#define XHCI_ZHAOXIN_TRB_FETCH	BIT_ULL(46)
++#define XHCI_ZHAOXIN_HOST	BIT_ULL(47)
```
Comment (hauke) [inline target/linux/bcm53xx/patches-6.1/180-usb-xhci-add-support-for-performing-fake-doorbell.patch:118]: Please move `XHCI_FAKE_DOORBELL` to the end too.

**Pull request merged**